literalai/client.py

```diff
@@ -25,6 +24,12 @@
 )
 from literalai.observability.thread import ThreadContextManager, thread_decorator
 
+from literalai.requirements import check_all_requirements
+
+LLAMA_INDEX_REQUIREMENT = ["llama-index>=0.10.58"]
```

Comment:
I am surprised mypy is not complaining since it is conditional and we use this function in the code. Why not move this logic in the client function?

Reply:
I was very surprised too. 
Much better to move it to inside the method itself. 

Side note, I noticed openai & mistralai instrumentations make sure of instrumenting only once with logic like: 
```
global is_mistralai_instrumented
    if is_mistralai_instrumented:
        return
```
I wonder whether we should do that with LlamaIndex.

Reply:
not sure, i think the llama index one is global by default?